app/js/elements/fxos-inspector.js

```diff
@@ -0,0 +1,485 @@
+        }));
+
+        this.keyPath.pop();
+      })();
+      break;
+    default:
+      break;
+  }
+
+  this.render();
+};
+
+proto._handleRemoveAttributeClick = function(target) {
+  var name = target.getAttribute('href');
+  this.target.removeNamedItem(name);
+
+  this.dispatchEvent(new CustomEvent('removeattribute', {
+    detail: JSON.stringify({
+      keyPath: this.keyPath,
+      expression: '[\'' + this.keyPath.join('\'][\'') + '\']',
+      name: name
+    })
+  }));
+
+  this.render();
+};
+
+proto._handleListItemClick = function(target) {
+  // If the link clicked has an `href` pointing
+  // to "../", go back to the previous level in
+  // the key path.
+  var href = target.getAttribute('href');
+  if (href === '../') {
+    this.keyPath.pop();
+  }
+
+  // Show "Create New Attribute" prompt.
+  else if (href === '#add-attribute') {
+    this.newAttributeName.open();
+  }
+
+  // Otherwise, push the next part of the key path.
+  else {
+    this.keyPath.push(href);
+  }
+
+  this.render();
+};
+
+/**
+ * Helper function for determining the type of
+ * page to render for the current target (editor
+ * or list).
+ */
+function renderTargetPage(target, isRoot) {
+  // If the target is not an object, assume that
+  // it is editable and render the editor page.
+  if (typeof target !== 'object') {
+    return renderTargetEditor(target);
+  }
+
+  // Otherwise, the target is an enumerable object
+  // and we can render the list page.
+  return renderTargetList(target, isRoot);
+}
+
+/**
+ * Helper for rendering the editor page.
+ */
+function renderTargetEditor(target) {
+  var page = document.createElement('section');
+  page.innerHTML =
+`<fxos-code-editor>${target}</fxos-code-editor>
+<section flex>
+  <gaia-button data-action="cancel">Cancel</gaia-button>
+  <gaia-button data-action="save">Save</gaia-button>
+</section>`;
+
+  return page;
+}
+
+/**
+ * Helper for rendering the list page.
+ */
+function renderTargetList(target, isRoot) {
+  var list = document.createElement('gaia-list');
+
+  if (!isRoot) {
+    list.appendChild(renderBackItem());
+  }
+
+  if (target instanceof window.NamedNodeMap) {
+    list.appendChild(renderAddAttributeItem());
+  }
+
+  getSortedProperties(target).forEach((property) => {
+    list.appendChild(renderTargetListItem(target, property));
+  });
+
+  var page = document.createElement('section');
+  page.innerHTML =
+`<section flex>
+  <gaia-text-input type="search" placeholder="Search Properties"></gaia-text-input>
+</section>
+${list.outerHTML}`;
+
+  return page;
+}
+
+/**
+ * Helper for rendering a list item.
+ */
+function renderTargetListItem(target, property) {
+  var value = target[property];
+  if (value instanceof window.Attr) {
+    value = `'${escapeHTML(value.value)}'`;
+  }
+
+  else if (typeof value === 'string') {
+    value = `'${escapeHTML(value)}'`;
+  }
+
+  var a = document.createElement('a');
+  a.href = property;
+  a.dataset.search = property.toLowerCase();
+  a.innerHTML =
```

Comment:
Yeah. I didn't want to stray too far from the standard font size of the gaia-list component to remain consistent. But, I think we can shrink it up a bit and it'll still look good.